### PR TITLE
Easi 2745/dataloaders task list sections

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
# EASI-2745

## Changes and Description

- Implement data loaders for multiple sections
   - Task List Sections
   - Plan Collaborator
   - Operational Solution Subtask
   
- Unit tests for each data loader

- Methodology
   -  I accomplished this by templating the changes and applying it for each object. You can see a fairly standard approach throughout the commits. 

## Future Potential data loaders
The following are not using data loaders, but might be queried for a model plan collection. It might be worth while to consider in this PR whether they should be ticketed to convert to a data loader or not as well.
1. Plan Documents
2. Is collaborator
3. Is Favorite
4.  crTDLS
5. prepareForClearance
6. nameHistory

## How to test this change
1. Run the application
2. run scripts/dev db:seed multiple times
3. Run the below query 
``` gql
query modelPlanCollection {
    modelPlanCollection(filter: INCLUDE_ALL) {
        id
        modelName
        discussions {
            content
            replies {
                content
                resolution
            }
        }
        basics{
            modelPlanID        
            id
            modelCategory
        }
        generalCharacteristics {
            id
        }
        participantsAndProviders {
            id
        }
        beneficiaries {
            id
        }
        opsEvalAndLearning {
            id
        }
        collaborators {
        id
        }
        operationalNeeds {
 
            id
            modelPlanID
            name
            key
            # section
            nameOther
            needed
            solutions(includeNotNeeded:true) {
                operationalSolutionSubtasks{
                    id
                    name
                    status
                }
                name
                key   
                needed
                nameOther
                pocEmail
                pocName     
                createdBy
                createdDts
            }
        }
        
    }
}
```

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [x] Added or updated tests for backend resolvers or other functions as needed.
- [x] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/storage/plan_general_characteristicsStore.go [hunk 1]
```diff
@@ -76,24 +76,23 @@ func (s *Store) PlanGeneralCharacteristicsGetByID(logger *zap.Logger, id uuid.UU
...
+func (s *Store) PlanGeneralCharacteristicsGetByModelPlanIDLOADER(logger *zap.Logger, paramTableJSON string) ([]*models.PlanGeneralCharacteristics, error) {
+	genCharSlice := []*models.PlanGeneralCharacteristics{}
 
-	statement, err := s.db.PrepareNamed(planGeneralCharacteristicsGetByModelPlanIDSQL)
+	stmt, err := s.db.PrepareNamed(planGeneralCharacteristicsGetByModelPlanIDLoaderSQL)
 	if err != nil {
 		return nil, err
 	}
-
 	arg := map[string]interface{}{
```
Comment (OddTomBrooks) [inline pkg/storage/plan_general_characteristicsStore.go:87]: NIT: If you prefer, there is a utility method to help make these string-map interfaces clearer:
CreateNamedIDQueryMap

File: pkg/graph/resolvers/operational_solution_subtask_test.go [hunk 2]
```diff
@@ -65,3 +75,49 @@ func (suite *ResolverSuite) TestOperationalSolutionSubtaskDelete() {
 	suite.NoError(err)
 	suite.EqualValues(numDeletedRows, 1)
 }
+
+func (suite *ResolverSuite) TestOperationalSolutionSubtaskDataLoader() {
+	needType := models.OpNKManageCd
+	plan := suite.createModelPlan("Plan For OpSolS 1")
+	need, err := suite.testConfigs.Store.OperationalNeedGetByModelPlanIDAndType(suite.testConfigs.Logger, plan.ID, needType)
+	suite.NoError(err)
+	sol1, _ := OperationalSolutionInsertOrUpdateCustom(suite.testConfigs.Logger, need.ID, "AnotherSolution", nil, suite.testConfigs.Principal, suite.testConfigs.Store)
```
Comment (OddTomBrooks) [inline pkg/graph/resolvers/operational_solution_subtask_test.go:84]: NIT: Personal preference, I tend to prefer more verbose names like "solution1" as I believe it keeps the code more readable. It's a very minor and obviously optional thing.

**Review by OddTomBrooks — APPROVED** [review]
Great work, Steven
